packages/amplify_flutter/lib/categories/amplify_auth_category.dart

```diff
@@ -115,27 +183,47 @@ class AuthCategory {
         : throw _pluginNotAddedException("Auth");
   }
 
+  /// Returns an [AuthUser] or plugin-specific subclass.
+  ///
+  /// Throws an [AuthException], or possibly a subclass such as [SignedOutException].
+  /// Other AuthException subclasses are possible as a result of network issues,
+  /// misconfiguration, internal service errors or other less common reasons.
   Future<AuthUser> getCurrentUser() {
     var request = AuthUserRequest();
     return plugins.length == 1
         ? plugins[0].getCurrentUser(request: request)
         : throw _pluginNotAddedException("Auth");
   }
 
+  /// Returns [<List<AuthUserAttribute>>] or a list of a plugin-specific subclass.
+  ///
+  /// Throws an [AuthException], or possibly a subclass such as [SignedOutException].
+  /// Other AuthException subclasses are possible as a result of network issues,
+  /// misconfiguration, internal service errors or other less common reasons.
   Future<List<AuthUserAttribute>> fetchUserAttributes({AuthUserAttributeOptions options}) {
     var request = AuthUserAttributeRequest(options: options);
     return plugins.length == 1
         ? plugins[0].fetchUserAttributes(request: request)
         : throw _pluginNotAddedException("Auth");
   }
 
+  /// Returns [AuthSession] or a plugin-specific subclass.
+  ///
+  /// Throws an [AuthException], or possibly a subclass such as [SessionExpiredException] or [SignedOutException].
+  /// Other AuthException subclasses are possible as a result of network issues,
+  /// misconfiguration, internal service errors or other less common reasons.
   Future<AuthSession> fetchAuthSession({AuthSessionOptions options}) {
```

Comment:
For all the APIs where we know `options` are used for some specific use cases, we should add them in their respective docs. `Options` in general is the most ambiguous parameter we have and is not really easily understood where and why one should use one.

Reply:
Agree with this.  I've added documentation for options in the SignUp API, and will add more once I fix the validationData/clientMetadata and get it into our official documentation (also once we figure out global signout).